test/integration/mongodb-handshake/mongodb-handshake.test.ts

```diff
@@ -0,0 +1,45 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+
+import {
+  Connection,
+  LEGACY_HELLO_COMMAND,
+  MongoServerError,
+  MongoServerSelectionError
+} from '../../mongodb';
+
+describe('MongoDB Handshake Node tests', () => {
+  let client;
+
+  context('when hello is too large', () => {
+    before(() => {
+      sinon.stub(Connection.prototype, 'command').callsFake(function (ns, cmd, options, callback) {
```

Comment:
Why is stubbing necessary?  And if we're stubbing, should these be unit? (my preference is no stubbing and leave as integration)

Reply:
The logic in our makeMetadata prevents me from constructing a metadata that's too large, so I'm using sinon to circumvent that. I still need a server, I'm invoking the real command method after I've made the 'too large' metadata document

Reply:
I'll be upfront and say I think this is unnecessary.  The changes you're proposing here are testing server behavior, not driver behavior.  If the server changes / removes the metadata limit, the driver doesn't care and we should continue to pass the prose tests (even if we silently pass them).

If we do care about checking whether or not the server throws if a document is too large, then we should directly test that using a `runCommand` with a manually constructed `hello`.  But I don't think we should test this.

Reply:
Oh good call on runCommand that would be better, I just didn't think of that

Reply:
Ah wait, building a hello isn't super straight forward, I'd have to limit the breadth of the server testing, by using the stubbing here I can cover auth and no-auth envs